tools/harness-simulation/harness/Thread_Harness/Sniffer/SimSniffer.py

```diff
@@ -0,0 +1,221 @@
+
+        devs = [SimSniffer(addressofDevice=addr, channel=11) for addr in devs]
+        print(devs)
+
+        return devs
+
+    @watched
+    def startSniffer(self, channelToCapture, captureFileLocation, includeEthernet=False):
+        self.channel = channelToCapture
+        self._local_pcapng_location = captureFileLocation
+        self._remote_pcap_location = os.path.join(
+            REMOTE_SNIFFER_OUTPUT_PREFIX,
+            ''.join(random.choice(string.ascii_letters + string.digits) for _ in range(16)) + '.pcap')
+
+        self._ssh = paramiko.SSHClient()
+        self._ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        remote_ip = self.ipaddr.split('@')[1]
+        self._ssh.connect(remote_ip, port=REMOTE_PORT, username=REMOTE_USERNAME, password=REMOTE_PASSWORD)
+
+        _, stdout, _ = self._ssh.exec_command(
+            'echo $$ && exec python3 %s -o %s -c %d' %
+            (os.path.join(REMOTE_OT_PATH, 'tools/harness-simulation/posix/sniffer_sim/sniffer.py'),
+             self._remote_pcap_location, self.channel))
+        self._remote_pid = int(stdout.readline())
+
+        self.log('local pcapng location = %s', self._local_pcapng_location)
+        self.log('remote pcap location = %s', self._remote_pcap_location)
+        self.log('remote pid = %d', self._remote_pid)
+
+        self.is_active = True
+
+    @watched
+    def stopSniffer(self):
+        if not self.is_active:
+            return
+        self.is_active = False
+
+        assert self._ssh is not None
+        self._ssh.exec_command('kill -s TERM %d' % self._remote_pid)
+        # wait to make sure the file is closed
+        time.sleep(3)
+
+        # truncate suffix from .pcapng to .pcap
+        local_pcap_location = self._local_pcapng_location[:-2]
+        with self._ssh.open_sftp() as sftp:
+            sftp.get(self._remote_pcap_location, local_pcap_location)
+        self._ssh.close()
+
+        cmd = [EDITCAP_PATH, '-F', 'pcapng', local_pcap_location, self._local_pcapng_location]
+        self.log('running editcap: %r', cmd)
+        subprocess.Popen(cmd).wait()
+        self.log('editcap done')
+
+        self._local_pcapng_location = None
+        self._ssh = None
+        self._remote_pcap_location = None
+        self._remote_pid = None
+
+    @watched
+    def setChannel(self, channelToCapture):
+        self.channel = channelToCapture
+
+    @watched
+    def getChannel(self):
+        return self.channel
+
+    @watched
+    def validateFirmwareVersion(self, device):
+        return True
+
+    @watched
+    def isSnifferCapturing(self):
+        return self.is_active
+
+    @watched
+    def getSnifferAddress(self):
+        return self.ipaddr
+
+    @watched
+    def globalReset(self):
+        pass
+
+    def log(self, fmt, *args):
+        try:
+            msg = fmt % args
+            print('%s - %s - %s' % ('SimSniffer', time.strftime('%b %d %H:%M:%S'), msg))
```

Comment:
class `OpenThread_Sim` has the `log` method as well, which is defined in the super class `OpenThreadTHCI`. It also directly uses `print` for logging.

Reply:
Currently use `self.log` in THCI and SI, because `@watch` decorator defined in `OpenThreadTHCI` directly uses `self.log`, and the methods in THCI and SI have to be decorated `@watch` for debug logging info unavoidably. I adopt `self.log` here for consistency.

In other cases, use `logging` library instead.